From 62fca66c937cfe5175e5257d4265a8db0e1270af Mon Sep 17 00:00:00 2001
From: tbalden <illespal@gmail.com>
Date: Thu, 27 Aug 2020 15:12:30 -0700
Subject: [PATCH 11/15] scripts/mkcompile_h: remove Clang's git URL from kernel
 version

---
 scripts/mkcompile_h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/mkcompile_h b/scripts/mkcompile_h
index 62d1597df27b..76c694b39529 100755
--- a/scripts/mkcompile_h
+++ b/scripts/mkcompile_h
@@ -76,7 +76,8 @@ UTS_TRUNCATE="cut -b -$UTS_LEN"
   echo \#define LINUX_COMPILE_BY \"`echo $LINUX_COMPILE_BY | $UTS_TRUNCATE`\"
   echo \#define LINUX_COMPILE_HOST \"`echo $LINUX_COMPILE_HOST | $UTS_TRUNCATE`\"
 
-  echo \#define LINUX_COMPILER \"`$CC -v 2>&1 | grep ' version ' | sed 's/[[:space:]]*$//'`\"
+  echo \#define LINUX_COMPILER \"`$CC -v 2>&1 | grep ' version ' | sed 's/clang/Clang/' \
+              | sed 's/(https..*)//g' | sed 's/[[:space:]]*$//'`\"
 ) > .tmpcompile
 
 # Only replace the real compile.h if the new one is different,
-- 
2.25.1

